src/main/java/com/ss/training/lms/controller/AdminAuthorController.java

```diff
@@ -39,4 +41,24 @@
 		return new ResponseEntity<Author>(author, status);
 	}
 
+	/**
+	 * @param author
+	 * @return
+	 */
+	@PutMapping(path = "/lms/admin/author")
+	public ResponseEntity<Author> updateAuthor(@RequestBody Author author) {
+		HttpStatus status = HttpStatus.BAD_REQUEST;
+		if (author == null || author.getAuthorName() == null || author.getAuthorName().length() > 45)

```

Comment:
Do we need to check the length here? the database itself has a rule to check the length of the string.

Reply:
Checking length here allows us to discern between a client error and a server error. If we pass a string that's too long to the database, it throws an SQLException, same as if we misspelled a query or misconfigured our connection.